examples/SharedMemory/PhysicsServerCommandProcessor.cpp

```diff
@@ -109,6 +110,47 @@ struct UrdfLinkNameMapUtil
 };
 
 
+class b3ThreadPool {
+public:
+	b3ThreadPool(const char *name = "b3ThreadPool") {
+		btThreadSupportInterface::ConstructionInfo info(name, threadFunction);
+		m_threadSupportInterface = btThreadSupportInterface::create(info);
+	}
+
+	~b3ThreadPool() {
+		delete m_threadSupportInterface;
+	}
+
+	const int numWorkers() const { return m_threadSupportInterface->getNumWorkerThreads(); }
+
+	void runTask(int threadIdx, btThreadSupportInterface::ThreadFunc func, void *arg) {
+		FunctionContext ctx = m_functionContexts[threadIdx];
```

Comment:
ctx is a local copy of the m_functionContexts[.], and this local variable goes out-of-scope while the threads are using it.
Please use a reference instead:
~~~~~~~~~~~~
	FunctionContext& ctx = m_functionContexts[threadIdx];
~~~~~~~~~~~~

Reply:
Thanks for catching this!